src/Functions/DateTimeTransforms.h

```diff
@@ -537,8 +537,12 @@ struct ToStartOfInterval<IntervalKind::Microsecond>
         else if (scale_multiplier > 1000000)
         {
             Int64 scale_diff = scale_multiplier / static_cast<Int64>(1000000);
-            if (t >= 0) [[likely]]
-                return t / microseconds / scale_diff * microseconds;
+            if (t >= 0) [[likely]] /// When we divide the `t` value we should round the result
+            {
+                bool is_ceiled = t / microseconds % scale_diff >= (5 * scale_diff / 10);
+                auto result = t / microseconds / scale_diff * microseconds;
+                return is_ceiled ? result + 1 : result;
+            }
```

Comment:
It seems like this way would be more efficient
```suggestion
            return (t / microseconds + scale_diff / 2) / scale_diff * microseconds;
```